traffic.lua

```diff
@@ -297,20 +284,32 @@ function courseplay:findaiTrafficCollisionTrigger(vehicle)
 	end;
 
 	local ret = false
-
+	local index = nil
+	
 	if vehicle.aiTrafficCollisionTrigger == nil then
-		local index = vehicle.i3dMappings.aiCollisionTrigger --getXMLString(xmlFile, "vehicle.aiTrafficCollisionTrigger#index");
+		if vehicle.i3dMappings.aiCollisionTrigger then		-- standard colli definition
+			index = vehicle.i3dMappings.aiCollisionTrigger
+		elseif vehicle.i3dMappings.trafficCollisionTrigger then		-- workaround GIANTS FS19 vehicle
+			index = vehicle.i3dMappings.trafficCollisionTrigger
+		elseif vehicle.i3dMappings.collisionTrigger then			-- workaround GIANTS FS19 vehicle
+			index = vehicle.i3dMappings.collisionTrigger
+		elseif vehicle.i3dMappings.aiTrafficTrigger then			-- workaround GIANTS FS19 vehicle
+			index = vehicle.i3dMappings.aiTrafficTrigger
+		elseif vehicle.i3dMappings.aiCollisionTriggerBig then			-- workaround GIANTS FS19 vehicle K105, K165
+			index = vehicle.i3dMappings.aiCollisionTriggerBig
+		elseif vehicle.i3dMappings.aiCollisionTriggerSmall then			-- workaround GIANTS FS19 vehicle K105, K165
+			index = vehicle.i3dMappings.aiCollisionTriggerSmall
+		end
 		if index then
 			local triggerObject = I3DUtil.indexToObject(vehicle.components, index);
 			if triggerObject then
 				vehicle.aiTrafficCollisionTrigger = triggerObject;
 			end;
 		end;
-	else
-		-- CpManager.trafficCollisionIgnoreList[vehicle.aiTrafficCollisionTrigger] = true; --add AI traffic collision trigger to global ignore list
 	end;
 	
 	if vehicle.aiTrafficCollisionTrigger == nil and getNumOfChildren(vehicle.rootNode) > 0 then
+		courseplay:debug(string.format("%s:findaiTrafficCollisionTrigger: no aiCollisionTrigger found in vehicle XML - trying alternative", nameNum(vehicle)), 3);	
```

Comment:
we have a courseplay.debugVehicle() to make this easier.

Reply:
Is it like this: courseplay.debugVehicle( 3, vehicle, "findaiTrafficCollisionTrigger: no aiCollisionTrigger found in vehicle XML - trying alternative") / channel 3 I guess...

Reply:
correct. And it can directly use a format string, you don't need string.format either. See examples all over the code.

Reply:
Understood. I just used the same I have seen already in this file.